File: CompilationTest/TernaryTest/results/TernaryTest.php
<?php

class TestClass {
    function getResult() {
        return true;
    }
    
    function testMethod() {
        print($this->getResult() ? "true" : "false" . "\n");
    }
}

try {
    $c = new TestClass();
    $c->testMethod();
} catch (Exception $err) {
    echo 'Exception: ' . $err->getMessage() . "\n";
}

File: CompilationTest/TernaryTest/results/TernaryTest.cpp
#include <iostream>

class TestClass {
  public:
    bool getResult() {
        return true;
    }
    
    void testMethod() {
        std::cout << (this->getResult() ? std::string("true") : std::string("false")) << std::endl;
    }

  private:
};

int main()
{
    try {
        TestClass c;
        c.testMethod();
    } catch(std::exception& err) {
        std::cout << "Exception: " << err.what() << '\n';
    }
    return 0;
}